### PR TITLE
Invites: updates to /accept API response payload

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request
Partly fixes an issue where accepting an invite becomes really slow once the user has a lot of sites already. The actual fix is done on the API side. This PR deals with changes on the response payload, and removes an unnecessary "full refresh" on the sites store to further improve performance.

* Update `/accept` payload expectations -- we will no longer receive a sites list after D62639-code is deployed.
* Use new endpoint version 1.3
* Replace unnecessary call to `requestSites()` with a call to `requestSite()` on the site that was recently joined by the user. h/t @tyxla for this suggestion

#### Testing instructions
- Apply D62639-code and sandbox your test site.

Browser A/User A:
1. Send an invite to User B.

Browser B/User B: 
1. Open the invite link from the email in a new tab, and replace `wordpress.com` with `calypso.locahost:3000`
2. You should be able to accept the invite without problems, and should get redirected to the accepted site's Calypso admin interface (or frontend, if a P2 site) upon accepting.

Related to #53454

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
Not approving simply because I haven't tested, but code-wise this looks awesome!

Thanks for the improvement! 🙌 

Noting that when we did the same improvement for the iframed Gutenberg editor in Calypso, it improved the loading time of the editor for many a12s by over 50%. That can really make a difference for users with many sites in their accounts.

**Comment by annemirasol** [comment]
Some e2e tests are failing because /accept v1.3 is not yet in prod. D62639-code needs to be deployed before these changes are merged.

**Comment by annemirasol** [comment]
> **This PR modifies the release build for wpcom-block-editor**
> 
> To test your changes on WordPress.com, run `install-plugin.sh wpcom-block-editor fix/slow-accept-invite` on your sandbox.
> 
> To deploy your changes after merging, see the documentation: PCYsg-l4k-p2


@tyxla 👋 Would you know why I'm getting this message? Is it because I changed the `/accept` endpoint version to 1.3?

I ran the provided `install-plugin.sh` command on my sandbox, and I'm seeing changes that aren't from this PR. e.g. 
```
-  var _block$attributes, _block$attributes2;
+  var _select$getActiveBloc;
```

**Comment by tyxla** [comment]
> @tyxla 👋 Would you know why I'm getting this message? Is it because I changed the `/accept` endpoint version to 1.3?

I think this is a false positive, you shouldn't be getting this message IMHO. @Automattic/team-calypso-platform could anyone take a look at this?

**Review by jsnajdr — APPROVED** [review]
Looks good to me. Good to merge after the 1.3 endpoint is deployed.

I also suggested in D62639 to return the one site that was added and do `receiveSite( data.site )` in Calypso, avoiding a REST roundtrip to request the site separately. But it's a nice-to-have idea, not meant to block anything.

**Comment by annemirasol** [comment]
> Looks good to me. Good to merge after the 1.3 endpoint is deployed.
> 
> I also suggested in D62639 to return the one site that was added and do `receiveSite( data.site )` in Calypso, avoiding a REST roundtrip to request the site separately. But it's a nice-to-have idea, not meant to block anything.

Good suggestion! API response in D62639 updated. I have also removed the separate call to `requestSite( siteId )`, since that information is now coming from the `/accept` response.

**Review by lezama — APPROVED** [review]
thanks for all the iterations and reviews!

**Review by jsnajdr — APPROVED** [review]
Looks good, I like both the back- and front-end patches :+1:

**Pull request merged**
